src/core/messages.ts

```diff
@@ -228,32 +228,48 @@ export class DefaultMessages
    * @param clientId The client ID of the user.
    * @param logger An instance of the Logger.
    */
-  constructor(roomId: string, realtime: Ably.Realtime, chatApi: ChatApi, clientId: string, logger: Logger) {
+  constructor(
+    roomId: string,
+    realtime: Ably.Realtime,
+    chatApi: ChatApi,
+    clientId: string,
+    logger: Logger,
+    initAfter: Promise<void>,
+  ) {
     super();
     this._roomId = roomId;
-    this._channel = getChannel(messagesChannelName(roomId), realtime);
-    addListenerToChannelWithoutAttach({
-      listener: this._processEvent.bind(this),
-      events: [MessageEvents.Created],
-      channel: this._channel,
+
+    this._channel = initAfter.then(() => {
+      const channel = getChannel(messagesChannelName(roomId), realtime);
+
+      addListenerToChannelWithoutAttach({
+        listener: this._processEvent.bind(this),
+        events: [MessageEvents.Created],
+        channel: channel,
+      });
+
+      // Handles the case where channel attaches and resume state is false. This can happen when the channel is first attached,
+      // or when the channel is reattached after a detach. In both cases, we reset the subscription points for all listeners.
+      channel.on('attached', (message) => {
+        this._handleAttach(message.resumed);
+      });
+      // Handles the case where an update message is received from a channel after a detach and reattach.
+      channel.on('update', (message) => {
+        if (message.current === 'attached' && message.previous === 'attached') {
+          this._handleAttach(message.resumed);
+        }
+      });
+
+      return channel;
     });
 
+    // catch this so it won't send unhandledrejection global event
```

Comment:
little nitpick with this and a couple other comments to keep case consistency :) 
```suggestion
    // Catch this so it won't send unhandledrejection global event
```